tidy3d/components/types.py

```diff
@@ -190,9 +190,9 @@ def __modify_schema__(cls, field_schema):
 Axis = Literal[0, 1, 2]
 Axis2D = Literal[0, 1]
 Shapely = BaseGeometry
-Vertices = Union[Tuple[Coordinate2D, ...], ArrayFloat2D]
 Vertices = constrained_array(ndim=2, dtype=float)
 PlanePosition = Literal["bottom", "middle", "top"]
+OperationType = Literal["union", "intersection", "difference", "symmetric_difference"]
```

Comment:
I guess `SetOperationType` might be a good option that indicates the use precisely.

Reply:
I think what I worry about is if (down the line) we have other set operations that use a different set of set operations (if that makes sense..)

Reply:
Yes, maybe `buffer`, `open`, and `close` could be added at some point. The more formal name would be `ClipOperationType`, because the usual "boolean" operations between polygons are generally called polygon clipping. `BooleanOperationType` is another option that might end up being used for other logical operations if we ever get those for other primitives too.

Reply:
the point of this type is to define what kind of set operations we can perform between polygons right? not necessarily to define all of the possible boolean operations or set operations, so I think ClipOperationType or even PolygonOperationType makes some sense maybe?